src/components/InventoryTable/EntityTableToolbar.js

```diff
@@ -237,35 +237,15 @@ const EntityTableToolbar = ({
    */
   const onRefreshDataInner = useCallback(
     (options) => {
+      console.log('OPSSSS', options);
       if (hasAccess) {
         onRefreshData(options);
         if (showTags && !hasItems) {
           dispatch(fetchAllTags(filterTagsBy, {}, getTags));
         }
       }
     },
-    [customFilters?.tags]
-  );
-
-  /**
-   * Function used to update data, it either calls `onRefresh` from props or dispatches `onRefreshData`.
-   * `onRefresh` function takes two parameters
-   *   * entire config with new changes.
-   *   * callback to update data.
-   * @param {*} config new config to fetch data.
-   */
-  const updateData = (config) => {
-    if (hasAccess) {
-      onRefreshDataInner(config);
-    }
-  };
-
-  /**
-   * Debounced `updateData` function.
-   */
-  const debouncedRefresh = useCallback(
```

Comment:
This is mainly the culprit as far as I remember, starting from line 247. 

These functions and callbacks were called when they shouldn't have been, due to the debouncing here. These changes in the table and the toolbar change that a little and consolidate everything to only call the ONE debounced refresh function in the EntityTable above.